### PR TITLE
feat(helm): adding kubeconfig flag

### PR DESCRIPTION
**Pull request by gardlt**
Closses-Issue: #2219

### CODE REVIEW

File: pkg/helm/environment/environment.go [hunk 1]
```diff
@@ -55,6 +57,8 @@ type EnvSettings struct {
 	Home helmpath.Home
 	// Debug indicates whether or not Helm is running in Debug mode.
 	Debug bool
+	// Kubeconfig is the local paht to the kubernetes config directory
```
Comment (technosophos) [inline pkg/helm/environment/environment.go]: s/paht/path/

**Comment by technosophos** [comment]
Thanks! This is overall looking good.

I'm a little concerned about the env var handling:

- We can't break backward compatibility and ignore $KUBECONFIG, but it looks like this breaks that
- There seems to be a new env var called HELM_KUBECONFIG, and I'm not sure what the intention for that is

Would you mind explaining how this part works?

**Comment by gardlt** [comment]
@technosophos so this should not break backwards compatibility. We will still be honoring the $KUBECONFIG var and default config file. if the flag is present than we override these.

HELM_KONFIG Thank you for catching this, I meant to get rid of the HELM_KUBECONFIG. i am removed

File: cmd/helm/helm.go [hunk 2]
```diff
@@ -47,6 +47,7 @@ var (
 	tlsEnable     bool   // enable TLS
 
 	kubeContext  string
+        kubeConfig   string
```
Comment (bacongobbler) [inline cmd/helm/helm.go]: looks like this file needs to be run against `go fmt`

**Comment by technosophos** [comment]
I think we're just waiting for this to pass the CircleCI build, right? I kicked it off again.

**Comment by technosophos** [comment]
Okay, all this needs is a run of `make docs` and a new commit. Can you please do that, @gardlt, and we'll get this merged for 2.6?

**Comment by gardlt** [comment]
@technosophos does this look correct.

File: pkg/helm/environment/environment.go [hunk 3]
```diff
@@ -46,13 +46,16 @@ type EnvSettings struct {
...
+	// KubeConfig is the name of the kubeconfig file.
+	KubeConfig string
 }
 
 // AddFlags binds flags to the given flagset.
 func (s *EnvSettings) AddFlags(fs *pflag.FlagSet) {
 	fs.StringVar((*string)(&s.Home), "home", DefaultHelmHome, "location of your Helm config. Overrides $HELM_HOME")
 	fs.StringVar(&s.TillerHost, "host", "", "address of Tiller. Overrides $HELM_HOST")
 	fs.StringVar(&s.KubeContext, "kube-context", "", "name of the kubeconfig context to use")
+	fs.StringVar(&s.KubeConfig, "kubeconfig", "", "path to kubeconfig file")
```
Comment (bacongobbler) [inline pkg/helm/environment/environment.go]: kubeconfig should override KUBECONFIG as do the other global flags e.g. $TILLER_NAMESPACE overrides `--tiller-namespace`, `--host` overrides $HELM_HOST etc.

**Review by bacongobbler — CHANGES_REQUESTED** [review]

File: pkg/helm/environment/environment.go (hunk 3 above)
Comment (gardlt) [reply pkg/helm/environment/environment.go]: so we want the $KUBECONFIG env variable to override the kubeconfig cli?

Comment (bacongobbler) [reply pkg/helm/environment/environment.go]: no, but we want to read the default as what's in $KUBECONFIG like we do with $HELM_HOST et al.

**Comment by bacongobbler** [comment]
ping, @gardlt are you able to continue working on this PR? If not that's okay.

**Comment by gardlt** [comment]
i can, did not see that there where conflicts i will resolve. did my latest 
change solve your comment above?

File: pkg/helm/environment/environment.go (hunk 3 above)
Comment (bacongobbler) [reply pkg/helm/environment/environment.go]: Would you mind updating this doc string to say "path to kubeconfig file. Overrides $KUBECONFIG" to maintain consistency with the rest of the flags?

**Comment by bacongobbler** [comment]
one minor nit then it looks good to go: https://github.com/kubernetes/helm/pull/2621#discussion_r137094221

**Comment by bacongobbler** [comment]
Also it looks like CircleCI is failing, so you'll need to fix it afterwards using `make docs` again.

**Comment by bacongobbler** [comment]
I'll have to give this a spin and see how this works. Code LGTM, though!

**Comment by thomastaylor312** [comment]
There hasn't been any more effort with this recently so I am going to bump it to 2.8. @gardlt if you can address the last outstanding comment on this soon, let me know and we can pull it back in to 2.7

**Comment by gardlt** [comment]
@thomastaylor312 i believe all issues have been addressed now. sorry for delay response.

**Review by bacongobbler — APPROVED** [review]

**Comment by bacongobbler** [comment]
LGTM. There's a breaking change introduced in pkg/kube but I think we can let this one slide in as it's a good change for the package overall, and we try to avoid breaking changes where possible, but don't explicitly ban them from happening.

Since this is a `size/large`, would someone else mind taking a look at this?

cc @adamreese in case you have some opinions on this part of the codebase.

File: pkg/helm/environment/environment.go (hunk 3 above)
Comment (thomastaylor312) [reply pkg/helm/environment/environment.go]: Looks like this doc string still needs to be updated as per @bacongobbler's comment

**Review by thomastaylor312 — CHANGES_REQUESTED** [review]
One tiny change needed and this should be good to go

File: pkg/helm/environment/environment.go (hunk 3 above)
Comment (gardlt) [reply pkg/helm/environment/environment.go]: Done

**Comment by gardlt** [comment]
My bad, fixing now.

**Review by thomastaylor312 — APPROVED** [review]

**Comment by thomastaylor312** [comment]
@bacongobbler This is good to merge when you would like

**Comment by thomastaylor312** [comment]
Oh, and thanks for the fix @gardlt!

**Pull request merged**